pkg/minikube/cluster/cluster.go

```diff
@@ -237,6 +238,12 @@ func StopHost(api libmachine.API) error {
 		return errors.Wrapf(err, "load")
 	}
 	console.OutStyle(console.Stopping, "Stopping %q in %s ...", cfg.GetMachineName(), host.DriverName)
+	if host.DriverName == constants.DriverHyperv {
+		glog.Infof("As there are issues with stopping Hyper-V VMs using API, trying to shut down using SSH")
+		if err := TrySSHPowerOff(host); err != nil {
+			return errors.Wrapf(err, "Unable to Power off cluster on %q using SSH", host.DriverName)
```

Comment:
This error message makes assumptions that the API cannot guarantee. Let TrySSHPowerOff supply the rich error message. This one should just say something like:

`errors.Wrap(err, "ssh power off")`

This way, when the error is ultimately handled by the program's main function, it will provide a clear causal chain from the root problem to the overall failures reminiscent of a NASA accident investigation:

`genesis: crashed: no parachute: G-switch failed: bad relay orientation`

Reply:
Roger that. I was not aware of how NASA accident investigations are done. Thanks for the insight!